target_hdfs/sinks.py

```diff
@@ -0,0 +1,72 @@
+"""hdfs target sink class, which handles writing streams."""
+
+from __future__ import annotations
+
+import os.path
+from pathlib import Path
+
+from target_parquet.sinks import ParquetSink
+
+from target_hdfs.utils.hdfs import (
+    read_most_recent_file,
+    upload_to_hdfs,
+)
+from target_hdfs.utils.parquet import get_parquet_files
+
+
+class CanNotUploadFileError(Exception):
+    """Can not upload file error."""
+
+
+class HDFSSink(ParquetSink):
+    """hdfs target sink class."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.logger.info(f"Target-hdfs initialized with {self.config}")
+        self.hdfs_destination_path = os.path.join(
+            self.config["hdfs_destination_path"], self.stream_name
+        )
+        self.skip_existing_files = self.config.get("skip_existing_files", False)
+        # Don't read the most recent file if partition_cols is set or if skip_existing_files is set
+        hdfs_file = (
+            read_most_recent_file(
+                self.hdfs_destination_path,
+                self.pyarrow_schema,
+                self.config.get("hdfs_block_size_limit"),
+            )
+            if not self.config.get("partition_cols") and not self.skip_existing_files
+            else None
+        ) or {}
+        self.hdfs_file_path = hdfs_file.get("path")
+        self.pyarrow_df = hdfs_file.get("content")
```

Comment:
Add some comment here that `self.pyarrow_df` is somehow used by super-class. At least I assume that it's used somehow.